orchestra/actions/install.py

```diff
@@ -31,38 +39,19 @@ def _run(self, args):
 
         pre_file_list = self._index_directory(tmp_root + orchestra_root, relative_to=tmp_root + orchestra_root)
 
-        start_time = time.time()
-
-        if self.from_binary_archives and self._binary_archive_exists():
-            logger.info("Fetching binary archive")
-            self._fetch_binary_archive()
-            logger.info("Extracting binary archive")
-            self._install_from_binary_archives()
-            source = "binary archive"
-
-            logger.info("Removing conflicting files")
-            self._remove_conflicting_files()
-            self.update_binary_archive_symlink()
-        elif not self.from_binary_archives or self.fallback_to_build:
-            self._install(args.quiet, args.test)
-            source = "build"
-
-            logger.info("Removing conflicting files")
-            self._remove_conflicting_files()
-
-            if self.build.component.skip_post_install:
-                logger.info("Skipping post install")
-            else:
-                self._post_install(args.quiet)
-
-            if args.create_binary_archives:
+        install_start_time = time.time()
+        if self.allow_binary_archive and self._binary_archive_exists():
+            self._install_from_binary_archive(args)
```

Comment:
The patch is hard to read, but I think you're no longer calling `update_binary_archive_symlink` after installing from binary archives. It is very important.
You have to be extra careful when doing this kind of rebase.

Reply:
I don't think you got the right line numbers for this comment, however, I do call `update_binary_archive_symlink` when a new binary archive is built. Is it necessary to call it regardless?
I was under the impression that its job is to update the link to the newly built binary archive

Reply:
I'm not talking about *building* a new binary archive, I'm talking about installing. This is why the comment is positioned here.

The old code was:
```python
        if self.from_binary_archives and self._binary_archive_exists():
            ...
            self.update_binary_archive_symlink()
         elif not self.from_binary_archives or self.fallback_to_build: 
            ...
             if args.create_binary_archives: 
                 self._create_binary_archive() 
             self.update_binary_archive_symlink() 
         else: 
             raise OrchestraException("Binary archive not found!") 
```

While now you're only updating symlinks if the archive is being built from source.
This is a bit subtle and opportunistic but it's necessary for the CI, in order to update symlinks after `next-branch` is promoted to `branch`, which is something we want even if we're installing from binary archives.
This is rather important.

Reply:
Ok I'll fix this right now, I didn't understand that the CI needed this behaviour

Reply:
Yeah, it wasn't obvious, maybe add a comment about this to avoid such a situation in the future.

Reply:
This commit should fix the issue https://github.com/revng/revng-orchestra/pull/9/files/e5f579a8d35264212095fd53515b96f9aadf928f..bdfa2922a4a21f2312fbaba4d2e7e2f94b58648b